proxy/hdrs/MIME.cc

```diff
@@ -2705,11 +2706,12 @@ mime_hdr_print(HdrHeap * /* heap ATS_UNUSED */, MIMEHdrImpl *mh, char *buf_start
   return 1;
 }
 
+namespace
+{
+template <int (*Char_transform)(int char_in)>
```

Comment:
Why template instead of passing in a function pointer? Looking ahead, I think it would be overall cleaner to use `std::transform` on the output buffer after writing the scheme and host. E.g. in `url_print` something like
```
if (normalize) {
  std::transform(pre, curr, &to_lower);
}
```